### PR TITLE
MMI radio upgrade

### PR DESCRIPTION
**Pull request by Crazylemon64**
I think I might have implemented things a tad ugly here, but I made sure to construct on existing systems when present.

A small framework for letting MMIs enable various forms of NanoUI access has also been added here - though right now, it's restricted only to the radio present when the upgrade is installed.

I poked at adding encryption key support, but as it stands that functionality is implemented separately across various types of radios, so I left the PR here, as a full refactor of that would take too long for not enough yield.

Alternative to #8953 

:cl:Crazylemon
rscadd: All MMI variants can now install an "MMI radio upgrade" in order to acquire radio capability when outside of any other chassis. It can be installed either directly on the MMI, or through an opened cyborg chassis. This radio can later be removed if desired by using a screwdriver on the MMI.
rscadd: MMIs can now pull up the direct interface of the radio instead of a single-toggle verb
rscadd: MMI radio control is now done via action button instead of via verb
rscdel: The radio MMI no longer exists as a distinct item
/:cl:

### CODE REVIEW

**Comment by DesolateG** [comment]
Nice!

**Comment by Spacemanspark** [comment]
Does this apply to positronics, as they are children of the MMI?

**Comment by DesolateG** [comment]
Yes

**Comment by Spacemanspark** [comment]
:+1:

**Comment by taukausanake** [comment]
So cyborgs without this upgrade can't talk on radio? What about AI units, do they have radio functionality? And is this the same as binary speak? I only ask because I don't know

**Comment by DesolateG** [comment]
No, the upgrade is for MMIs, allowing them radio capability in cases where they find themselves without a body or chassis, such as being detonated

**Comment by davidchan13** [comment]
Works for posibrains too? Woohoo!

**Comment by Keilden** [comment]
So could a IPC go to robotics and have them take out his Posibrain and upgrade it with a radio? Then put it back so that hey can talk if they get debrained later?

**Comment by davidchan13** [comment]
Very difficult to accidentally debrain a IPC, as the chest has to be surgeried or destroyed somehow. I do believe that Posibrain/MMI speakers are disabled by default how ever.

**Comment by Crazylemon64** [comment]
@Keilden Yes, that is possible - but you don't need the radio upgrade handy to turn on the speakers, that's just something you can do by hand once the posibrain's removed. And if the speakers are off, you still cannot use an integrated radio, so it can't be used to scream bloody murder if someone needs to brain-nap an IPC.

File: code/modules/mob/living/carbon/brain/MMI.dm [hunk 1]
```diff
@@ -116,31 +154,54 @@
 	held_brain.forceMove(dropspot)
 	held_brain = null
 
+/obj/item/mmi/proc/become_occupied(var/new_icon)
```
Comment (Crazylemon64) [inline code/modules/mob/living/carbon/brain/MMI.dm]: this one's kinda icky but it handles bundling action icon updating

**Pull request merged**